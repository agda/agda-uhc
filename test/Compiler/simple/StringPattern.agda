-- test for pattern matching on Strings
module StringPattern where

open import Common.IO
open import Common.Unit
open import Common.String

f : String â†’ String
f "bla" = "found-bla"
f x = x

-- expected:
-- no-bla
-- found-bla
main : IO Unit
main =
  putStrLn (f "no-bla") ,,
  putStrLn (f "bla")
