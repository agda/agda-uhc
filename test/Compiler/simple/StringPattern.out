EXECUTED_PROGRAM

ret > ExitSuccess
out > no-bla
out > found-bla
out >
